MAUI/DataForm/Working-with-DataForm.md

```diff
@@ -0,0 +1,781 @@
+
+{% tabs %}
+{% highlight C# %}
+
+    [Display(AutoGenerateField = false)]
+    public int ID { get; set; } 
+    
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+In the following code, the `DataFormItem` generation for the `MiddleName` property is canceled by setting the `Cancel` property to true.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem.FieldName == "MiddleName")
+        {
+            e.Cancel = true;
+
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Changing property settings
+
+You can change the property of `DataFormItem` in the `GenerateDataFormItem` event.
+
+Here, `Salary` data field is restricted from being edited in the data form.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "MiddleName")
+            {
+                e.DataFormItem.IsReadOnly = true;
+            }
+        }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+#### Changing DataFormItem visibility
+
+You can change the `DataFormItem` visibility by using the `IsVisible` property in the `DataFormItem`.
+
+Here, `LastName` data field will be hidden.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "LastName")
+            {
+                e.DataFormItem.IsVisible = false;
+            }
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Setting watermark
+
+You can display the watermark in the editor by defining the display attribute or using the`GenerateDataFormItem` event.
+
+#### Using attribute
+
+You can show the watermark in the editor by setting the `Prompt` in display attribute.
+
+{% tabs %}
+{% highlight C# %}
+
+    [Display(Prompt = "Enter middle name")]
+    public string MiddleName { get; set; }
+    
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+You can show the watermark in the editor by using the `PlaceholderText` property in `DataFormItem`.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "Description")
+            {
+                e.DataFormItem.PlaceholderText = "Enter description";
+            }
+        }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+## Setting watermark color
```

Comment:
## Changing watermark color for autogenerated editor 4#

You can display the color for the watermark in the editor by using `GenerateDataFormItem` event when Autogerate true

Reply:
title and content updated